### ISSUE TITLE
Migrating formly v5 to v6 in Angular 16

### ISSUE THREAD

**Issue by triadiprabowo** [ngx-formly/ngx-formly#3995]
**Question**

Hi,

I upgraded ngx-formly to version 6 in Angular 16, but it’s not working out of the box. The `required` html attribute is set correctly in DOM.

However, when I check the form control object, the `validator` property is null (this used to have function in v5), and the form stays valid even when the field is dirty and empty.

Is anyone else experiencing the same issue with ngx-formly v6 in Angular 16?

Problem:
1. Pre-defined validator is not set correctly (`required`, `pattern`, etc.)
2. `formControl` stays valid even though it is `dirty` (this is due to `validator` property is null)

Context:
1. `templateOptions.required = true;` (I know `templateOptions` still works in v6 as it's an alias of `props`)
2. I am using custom wrapper and custom field type

![Screenshot_2024-09-26_at_12 45 10 (1)](https://github.com/user-attachments/assets/a17d3c54-356e-4d27-b551-2a7f956bc835)

In my JSON field configuration I have this pre-defined:
`ff.expressions['templateOptions.required'] = (field:IExtraFieldConfigurationObject) => {
  return field.templateOptions.required;
};`

Do you know what can cause `validator` object set to null in `formControl` ?

Adding this workaround works in my super class of extended field type:
```
const validators:ValidatorFn[] = [];

    if (this.props.required) {
      validators.push(Validators.required);
    }

    if (this.props.pattern) {
      validators.push(Validators.pattern(this.props.pattern));
    }

    if (!_.isUndefined(this.props.maxLength)) {
      validators.push(Validators.maxLength(this.props.maxLength));
    }

    if (!_.isUndefined(this.props.minLength)) {
      validators.push(Validators.maxLength(this.props.minLength));
    }

    if (!_.isUndefined(this.props.max)) {
      validators.push(Validators.max(this.props.max));
    }

    if (!_.isUndefined(this.props.min)) {
      validators.push(Validators.max(this.props.max));
    }

    this.formControl?.setValidators(validators);
    this.formControl?.updateValueAndValidity({emitEvent: false});
```

Thank you and have a good day!

**Comment by aitboudad** [comment]
Hi, without a reproduction its difficult to know exactly what causes the issue.
if you can provide an example that we can check that would be helpful.

**Comment by aitboudad** [comment]
Please feel free to leave a comment once you've a got a reproduction example so we can check it further on our side and provide a fixed if needed.

Thank you!

**Comment by vijhhh2** [comment]
I have the same issue. I have recently upgraded  form v5 to v6. In my case we are setting up templateoptions.required = () => { function which returns true or false }. In UI it shows "*" correctly. But If I check the validator for the underlying control it set to null. Also I am able to submit form successfully which it should not be. Not sure what's happening. Right now I am not able to reproduce outside of my app. But this is a bug. @triadiprabowo apart from your suggestion did you try anything to fix this problem. I have tried everything but nothing worked for me.

Formly Version:
"@ngx-formly/core": "^6.3.12"
"@ngx-formly/material": "^6.3.12"

Angular version
"@angular/core": "^18.1.1",
